sdk/core/Azure.Core/src/ResourceType.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Structure representing a resource type.
+    /// </summary>
+    public readonly struct ResourceType : IEquatable<ResourceType>
+    {
+        private const char Separator = '/';
```

Comment:
nit: we define this in ResourceIdentifier as well - could we make it internal and define it only once?